### DISCUSSION TITLE
File system becomes read-only after resizing disk in Proxmox

### DISCUSSION THREAD

**Discussion by eugenefvdm** [TheGrandWazoo/freenas-proxmox#199] [Q&A] [answered]
When I use Proxmox VE 8.2.4 and earlier versions to resize the ZFS over iSCSI disk, the Linux operating system (e.g. Ubuntu) disk becomes read-only.

Here is an example:

![Screenshot 2024-10-26 at 01 10 04](https://github.com/user-attachments/assets/b3bc1139-257c-4152-9097-69859bf7d331)

If I resize that Ubuntu VM ext4 disk with 2 GB, the disk will crash and become read-only.

The solution is to reboot the server or to only do it when the server is offline.

However, I can categorically state on Local-LVM volumes, and LVM volumes connected via RAID hardware, I don't have the same problem.

Any tips?

**Comment by TheGrandWazoo** [comment, marked as answer]
Sorry this sat without a reply for so long. This isn't specific to this plugin — it's a general behavior of resizing a live-attached iSCSI LUN. When Proxmox extends the zvol, the guest's SCSI layer doesn't automatically notice the new size; ext4 (and similar filesystems) can see a mismatch between what the partition table/filesystem thinks the disk size is and what the block device now reports, and remount read-only as a safety measure. Local-LVM and hardware-RAID LVM don't hit this because there's no separate SCSI-rescan step involved.

Workaround: before growing the partition/filesystem in the guest, trigger a SCSI rescan first (on Linux: `echo 1 > /sys/class/scsi_device/*/device/rescan` for the affected device) so the guest kernel picks up the new size before you resize the partition table. That avoids the reboot-required path.

This applies the same way under both v2.x and v3.0 since it's inherent to iSCSI + online resize, not something the plugin controls.
